GNUmakefile

```diff
@@ -1,6 +1,7 @@
 SHELL = bash
 PROJECT_ROOT := $(patsubst %/,%,$(dir $(abspath $(lastword $(MAKEFILE_LIST)))))
 THIS_OS := $(shell uname | cut -d- -f1)
+THIS_ARCH := $(shell uname -a | cut -d" " -f9)
```

Comment:
While this may work on s390x, on macOS I get:

```
$ uname -a | cut -d" " -f9
Feb
```

and on Ubuntu amd64 I get:

```
$ uname -a | cut -d" " -f9
18:56:26
```

So trying to parse this doesn't seem robust enough... is there a better uname flag we could use here like `uname -p`?

Reply:
`uname -p` seems to work for Ubuntu, but not Debian with a 4.19.x kernel.

```
$ uname -p
s390x
$ cat /etc/issue
Ubuntu 20.04.1 LTS \n \l
```
Debian:
```
$ uname -p
unknown
$ cat /etc/issue
Debian GNU/Linux 10 \n \l

$ uname -a
Linux lpar08 4.19.0-12-s390x #1 SMP Debian 4.19.152-1 (2020-10-18) s390x GNU/Linux
```
Seems `uname -m` is the way to go. Will update the PR.